### PR TITLE
Fix change event on the select element

### PR DESCRIPTION
**Pull request by jingyu9575**
Sorry, I think I missed the last message. 

This commit makes the last section only show when the third option is selected.

By the way I cannot see any text on the settings page, because the English translation file is incomplete. The default language is set to "en" in manifest.json:

    "default_locale": "en",

so only the English file will be used in other locales without translation file.

### CODE REVIEW

File: src/manifest.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
   "optional_permissions": [
   "bookmarks",
   "downloads",
-  "browserSettings",
+  "browserSettings"
```
Comment (jingyu9575) [inline src/manifest.json:64]: Well, this extra comma after the end of the last value is not allowed by JSON, and Firefox will not load the extension if manifest.json is invalid. I'd say the JSON spec is too strict with commas.

File: src/html/Options.html [hunk 2]
```diff
@@ -84,5 +84,6 @@ <h3 data-i18n="local_file_saving"></h3>
 
       <script src='/js/extra-permission.js'></script>
       <script src='/js/i18n.js'></script>
+      <script src='/js/Options.js'></script>
```
Comment (jingyu9575) [inline src/html/Options.html:87]: The JS needs to be loaded here.

File: src/html/Options.html [hunk 3]
```diff
@@ -70,7 +70,7 @@ <h5 data-i18n="newtab_revoked"></h5>
                        
 
                    
-                    <div id="admDivCheck" style="display:none;"></div>
+                    <div id="admDivCheck" style="display:none;">
```
Comment (jingyu9575) [inline src/html/Options.html:73]: `<div ......>` is the beginning of the div element, and `</div>` is the ending. This div needs to contains the 6 elements below, so the `</div>` must be moved after them.

File: src/html/Options.html [hunk 4]
```diff
@@ -20,7 +20,7 @@ <h1 data-i18n="settings-title"></h1>
               <h3 data-i18n="option_selection"></h3>
               <form>
                   <div>
-                      <select id="type" name="type_select" onchange="admSelectCheck(this);">
```
Comment (jingyu9575) [inline src/html/Options.html:23]: Unfortunately Firefox extensions are subject to [Content Security Policy (CSP) restrictions](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_Security_Policy). When Firefox developed WebExtensions following the Chrome extensions API, this behavior is taken from Chrome extensions. 

Basically extensions cannot run JS code outside JS files. This `admSelectCheck(this);` is JS code but it is in the HTML file, so it is forbidden. The change event must be written in another way (see the JS file).

The original answer works in regular websites, but not in extensions.

File: src/js/Options.js [hunk 5]
```diff
@@ -6,18 +6,13 @@ const options = {
 
 };
 
-let admOptionValue = document.getElementById("local").value;
```
Comment (jingyu9575) [inline src/js/Options.js:9]: (I originaly suggested to use `function admSelectCheck(nameSelect) {` in this line. To workaround CSP, this function is written in another way.)

File: src/js/Options.js [hunk 6]
```diff
@@ -6,18 +6,13 @@ const options = {
...
-    if (nameSelect) {
-        admOptionValue = document.getElementById("local").value;
-        if (admOptionValue == nameSelect.value) {
-            document.getElementById("admDivCheck").style.display = "block";
-        }
-        else {
-            document.getElementById("admDivCheck").style.display = "none";
-        }
+let typeSelect = document.getElementById("type")
+typeSelect.addEventListener("change", function () {
```
Comment (jingyu9575) [inline src/js/Options.js:10]: First, the `<select>` is found with `getElementById`. To add an event handler, `addEventListener` is used. The first argument `"change"` is the event type (onchange), and the second argument is a function. This function is called whenever the element is changed. 

This statement is a function call: `typeSelect.addEventListener("change", ......);`, and the handler is a new function `function () { ...... }`. The function does not need a name.

File: src/js/Options.js [hunk 7]
```diff
@@ -6,18 +6,13 @@ const options = {
...
+    let admOptionValue = document.getElementById("local").value;
```
Comment (jingyu9575) [inline src/js/Options.js:11]: This line is where the original answer missed a `let` keyword.

File: src/manifest.json (hunk 1 above)
Comment (xerta555) [reply src/manifest.json:64]: Yeah, it indeed, i realized it later, and i had corrected the error, and yes the error is unforgivable at the JSON level.

File: src/html/Options.html (hunk 2 above)
Comment (xerta555) [reply src/html/Options.html:87]: Effectivly, i don't remember me why i removed the Options.js file loading in my html file....

File: src/html/Options.html (hunk 3 above)
Comment (xerta555) [reply src/html/Options.html:73]: Yes, also fixed after pushed my last changes on the repo.

File: src/html/Options.html (hunk 4 above)
Comment (xerta555) [reply src/html/Options.html:23]: Noted.

**Comment by xerta555** [comment]
> Sorry, I think I missed the last message.

Np.

> This commit makes the last section only show when the third option is selected.

> By the way I cannot see any text on the settings page, because the English translation file is incomplete. The default language is set to "en" in manifest.json:

> "default_locale": "en",

> so only the English file will be used in other locales without translation file.

I will fix it today, thank you again

**Pull request merged**